crates/zed/src/languages/haskell/highlights.scm

```diff
@@ -0,0 +1,632 @@
+;; Copyright 2022 nvim-treesitter
+;;
+;; Licensed under the Apache License, Version 2.0 (the "License");
+;; you may not use this file except in compliance with the License.
+;; You may obtain a copy of the License at
+;;
+;;     http://www.apache.org/licenses/LICENSE-2.0
+;;
+;; Unless required by applicable law or agreed to in writing, software
+;; distributed under the License is distributed on an "AS IS" BASIS,
+;; WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+;; See the License for the specific language governing permissions and
+;; limitations under the License.
+; ----------------------------------------------------------------------------
+; Parameters and variables
+; NOTE: These are at the top, so that they have low priority,
+; and don't override destructured parameters
+(variable) @variable
+
+(pat_wildcard) @variable
+
+(function
+  patterns:
+    (patterns
+      (_) @variable.parameter))
+
+(exp_lambda
+  (_)+ @variable.parameter
+  "->")
+
+(function
+  infix:
+    (infix
+      lhs: (_) @variable.parameter))
+
+(function
+  infix:
+    (infix
+      rhs: (_) @variable.parameter))
+
+; ----------------------------------------------------------------------------
+; Literals and comments
+(integer) @number
+
+(exp_negation) @number
+
+(exp_literal
+  (float)) @number.float
+
+(char) @character
+
+(string) @string
+
+(con_unit) @string.special.symbol ; unit, as in ()
+
+(comment) @comment
```

Comment:
The `highlights.scm` file had another bit for comments that I removed because it made the comments not render correctly, I am not sure what the purpose of that was but here's a link to the removed lines anyway:

https://github.com/nvim-treesitter/nvim-treesitter/blob/master/queries/haskell/highlights.scm#L621-L623

Reply:
Thanks for the link! I think in general we'd prefer to remove commented out highlights rather than keep them, so this was great :)